apps/onboarding-functions/pom.xml

```diff
@@ -13,12 +13,12 @@
 
   <properties>
     <compiler-plugin.version>3.11.0</compiler-plugin.version>
-    <maven.compiler.release>11</maven.compiler.release>
+    <maven.compiler.release>17</maven.compiler.release>
```

Comment:
consider to put these kind of changes into their own PRs in the future otherwise they make more difficult to track errors introduced with PRs